packages/block-editor/src/components/block-list/block.native.js

```diff
@@ -23,6 +28,66 @@ import BlockEdit from '../block-edit';
 import BlockInvalidWarning from './block-invalid-warning';
 import BlockMobileToolbar from '../block-mobile-toolbar';
 
+function BlockForType( {
+	attributes,
+	clientId,
+	contentStyle,
+	getBlockWidth,
+	insertBlocksAfter,
+	isSelected,
+	mergeBlocks,
+	name,
+	onBlockFocus,
+	onCaretVerticalPositionChange,
+	onChange,
+	onDeleteBlock,
+	onReplace,
+	parentWidth,
+	wrapperProps,
+} ) {
+	const defaultColors = useEditorFeature( 'color.palette' ) || [];
+
+	return (
+		<GlobalStylesContext.Consumer>
+			{ ( globalStyle ) => {
+				const mergedStyle = getMergedGlobalStyles(
+					globalStyle,
+					wrapperProps.style,
+					attributes,
+					defaultColors
+				);
+
+				return (
+					<GlobalStylesContext.Provider value={ mergedStyle }>
+						<BlockEdit
+							name={ name }
+							isSelected={ isSelected }
+							attributes={ attributes }
+							setAttributes={ onChange }
+							onFocus={ onBlockFocus }
+							onReplace={ onReplace }
+							insertBlocksAfter={ insertBlocksAfter }
+							mergeBlocks={ mergeBlocks }
+							onCaretVerticalPositionChange={
+								onCaretVerticalPositionChange
+							}
+							// Block level styles
+							wrapperProps={ wrapperProps }
+							// inherited styles merged with block level styles
+							mergedStyle={ mergedStyle }
+							clientId={ clientId }
+							parentWidth={ parentWidth }
+							contentStyle={ contentStyle }
+							onDeleteBlock={ onDeleteBlock }
```

Comment:
Seems like all of these props have the same name as destructured ones. WDYT about something like:

```js
function BlockForType( {
  attributes,
  getBlockWidth,
  wrapperProps,
  ...restProps
} )
```
```js
<BlockEdit
  {...restProps}
  // Block level styles
  wrapperProps={ wrapperProps }
  // inherited styles merged with block level styles
  mergedStyle={ mergedStyle }
/>
```

Reply:
Although this would be good because it'll remove some lines, sometimes I feel like when doing this it's hard to track props when checking/refactoring code or might cause unexpected results. Still, I tried your suggestion and I started to get issues with some props and its references... 😅

Reply:
Yeah, you're right. Since we don't have typing it would be good to leave it as it is for future refactors :)